### PR TITLE
disabling inspect and default pattern/timestamp buttons when using async data sources

### PR DESCRIPTION
**Pull request by paulstn**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/event_analytics/explorer/sidebar/field.tsx [hunk 1]
```diff
@@ -37,6 +40,7 @@ interface IFieldProps {
   showTimestampOverrideButton: boolean;
   isFieldToggleButtonDisabled: boolean;
   onToggleField: (field: IField) => void;
+  tabId: any;
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/sidebar/field.tsx:43]: string?

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ interface ISidebarProps {
   handleOverrideTimestamp: (timestamp: IField) => void;
   storedExplorerFields: IExplorerFields;
   setStoredExplorerFields: (explorer: IExplorerFields) => void;
+  tabId: any;
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/sidebar/sidebar.tsx:37]: use string type

**Review by mengweieric — APPROVED** [review]

File: public/components/event_analytics/explorer/sidebar/field.tsx (hunk 1 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/sidebar/field.tsx:43]: will modify

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx (hunk 2 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/sidebar/sidebar.tsx:37]: will modify

**Review by ps48 — APPROVED** [review]

**Comment by ps48** [comment]
@paulstn Can you look into the test errors?

**Pull request merged**